### DISCUSSION TITLE
Documenting how to use the Admin theme confirmDialog

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#9191] [General]
Took me a while to figure this out so I thought I'll document it.

If you wish to trigger a confirm modal popup, you can use  the existing JavaScript function in [admin.js]( https://github.com/OrchardCMS/OrchardCore/blob/1ec2e9898ff37e49d4224732c8a3e0acb5ed2fa6/src/OrchardCore.Themes/TheAdmin/Assets/js/admin.js#L1)

Steps:

1. For the older version (e.g. the current master branch) add a `itemprop` attribute

2. For the current version (e.g. the current dev branch) add a `data-url-af` attribute

3. To perform a `GET` to the url specified in the `href` attribute use `RemoveUrl` e.g. `data-url-af="RemoveUrl"`

4. To perform a `POST` to the url specified in the `href` attribute use `RemoveUrl UnsafeUrl` e.g. `data-url-af="RemoveUrl UnsafeUrl"`

5. To pass your custom message, use the `data-message` attribute. e.g. `data-message="my custom message"`

For example:

```html
<a
    asp-action="Delete"
    asp-route-id="@Model.Id"
    asp-area="OrchardCore.MyModule"
    asp-controller="MyController"
    itemprop="RemoveUrl UnsafeUrl"
    data-url-af="RemoveUrl UnsafeUrl"
    data-message="Hello World"
    class="btn btn-danger btn-sm">@T["Delete"]</a>
```

**Comment by hishamco** [comment]
> Took me a while to figure this out

If you filed an issue you will get the answer ;) but it's good you raise this. I'm not sure where we can document this

/cc @agriffard

**Reply by weirdyang** [reply]
It's fine,it's more beneficial for me to try and figure out the solution than expect to be given it. Helps me understand the entire project better too.
